dotcom-rendering/src/web/components/Liveness.importable.tsx

```diff
@@ -199,28 +199,35 @@ export const Liveness = ({
 			numHiddenBlocks > 0 ? `(${numHiddenBlocks}) ${webTitle}` : webTitle;
 	}, [numHiddenBlocks, webTitle]);
 
-	if (onFirstPage && topOfBlog) {
-		const observer = new window.IntersectionObserver(
-			([entry]) => {
-				if (entry.isIntersecting) {
-					entry.target.classList.add('in-viewport');
-					revealPendingBlocks();
-					setNumHiddenBlocks(0);
-					setShowToast(false);
-					return;
-				}
-				entry.target.classList.remove('in-viewport');
-			},
-			{
-				root: null,
-				// A margin makes more sense because the element we're
-				// observing (topOfBlog) has no height
-				rootMargin: '100px',
-			},
-		);
+	useEffect(() => {
```

Comment:
I added a useEffect here as an optimisation to prevent the observer being set multiple times

Reply:
Should we make sure that we have a `return` statement that disconnects it, as per hooks specs?

Should we leverage `useOnce`, if we truly want this to only run once? It feels like currently this would run each time the value of `onFirstPage` changes.

Reply:
> Should we make sure that we have a `return` statement that disconnects it, as per hooks specs?

Agreed. Great spot. I've added this.

> 
> Should we leverage `useOnce`, if we truly want this to only run once? It feels like currently this would run each time the value of `onFirstPage` changes.

Rerunning if/when `onFirstPage` changes is the intended behaviour. In reality this value is static but the component might rerender for other reasons (these could change in future) and so keeping the logic makes sense. I think `useOnce` is best for when you _really_ only want code to run once, like incrementing a value.